### PR TITLE
fixed streamline benchmarks

### LINKED ISSUE #1094: Benchmark for streamline broken
(This issue tracker is only for bug reports or feature requests, if this is neither, please choose appropriate channel from http://bluebirdjs.com/docs/support.html)

Please answer the questions the best you can:

1) What version of bluebird is the issue happening on?
HEAD @ c0d4472cecd523c2f9d4805a23d87be3cfe03b41

2) What platform and version? (For example Node.js 0.12 or Google Chrome 32)
Platform info:
Linux 3.13.0-85-generic x64
Node.JS 6.1.0
V8 5.0.71.35

3) Did this issue happen with earlier version of bluebird?
N/A
(Write description of your issue here, stack traces from errors and code that reproduces the issue are helpful)

Steps to reproduce --
1) git clone bluebird
2) npm install
3) ./bench doxbee

Stack trace --

```
benchmarking ./doxbee-sequential/streamline-callbacks.js
module.js:440
    throw err;
    ^

Error: Cannot find module 'streamline/lib/callbacks/runtime'
    at Function.Module._resolveFilename (module.js:438:15)
    at Function.Module._load (module.js:386:25)
    at Module.require (module.js:466:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/usr/local/google/home/gunasekaran/code/bluebird/benchmark/doxbee-sequential/streamline-callbacks.js:1:140)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
benchmarking ./doxbee-sequential/streamline-generators.js
module.js:440
    throw err;
    ^

Error: Cannot find module 'streamline/lib/generators/runtime'
    at Function.Module._resolveFilename (module.js:438:15)
    at Function.Module._load (module.js:386:25)
    at Module.require (module.js:466:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/usr/local/google/home/gunasekaran/code/bluebird/benchmark/doxbee-sequential/streamline-generators.js:1:144)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
```

```
➜  bluebird git:(master)   ls benchmark/doxbee-sequential/streamline*
benchmark/doxbee-sequential/streamline-callbacks.js  benchmark/doxbee-sequential/streamline-generators.js  benchmark/doxbee-sequential/streamline._js
```

### PR DESCRIPTION
**Pull request by bjouhier**
This PR fixes #1094. I have added the missing reference in benchmark/package.json.

I have also upgraded streamline to 2.0 and recompiled the files.

The streamline-callbacks bench is slower than before (and it is now slower than streamline-generators). This is because the streamline transform has been reimplemented with babel and the new transform uses regenerator instead of the original streamline algorithm.

### CODE REVIEW

**Comment by benjamingr** [comment]
Thanks! Is the code with all the switch/case the actual source code or a generated output?

**Comment by bjouhier** [comment]
The source is the `streamline._js` file. The file with the  big switch/case is produced by regenerator. 

The old callbacks algorithm was generating code that looked a bit better and was closer to the original source, but I got lazy when I reimplemented with babel and I just applied the regenerator transform on top of streamline's generators transform. Anyway, every JS engine supports generators these days...

**Comment by benjamingr** [comment]
LGTM

File: benchmark/doxbee-sequential-errors/streamline-callbacks.js [hunk 1]
```diff
@@ -1,43 +1,98 @@
...
-                  if (!file) {
-                    splitPath = idOrPath.split("/");
-                    fileName = splitPath[(splitPath.length - 1)];
+'use strict';
+
+var regeneratorRuntime = typeof require === 'function' ? require('streamline-runtime/lib/callbacks/regenerator') : Streamline.require('streamline-runtime/lib/callbacks/regenerator');
+
+var _streamline = typeof require === 'function' ? require('streamline-runtime/lib/callbacks/runtime') : Streamline.require('streamline-runtime/lib/callbacks/runtime');
+
+var _filename = '/Users/bruno/dev/bluebird/benchmark/doxbee-sequential-errors/streamline._js';
```
Comment (lvivski) [inline benchmark/doxbee-sequential-errors/streamline-callbacks.js]: Shouldn't it be relative, not hardcoded? There are several other places throughout the code that have it hardcoded

Comment (bjouhier) [inline benchmark/doxbee-sequential-errors/streamline-callbacks.js]: That's right. It won't do much harm as it is only used to reconstruct stack frames.

I've entered an issue: Sage/streamlinejs#339

**Comment by bambielli** [comment]
It appears that streamline-callbacks.js is now using a deprecated module. I get the following when I run your code:

```
The regenerator/runtime module is deprecated; please import regenerator-runtime/runtime instead.
```

**Comment by bjouhier** [comment]
Should be ok now.

**Pull request merged**